transforms/sort-comp.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Reorders React component methods to match the [ESLint](http://eslint.org/)
+ * [react/sort-comp rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md),
+ * specifically with the [tighter constraints of the Airbnb style guide](https://github.com/airbnb/javascript/blob/6c89f9587f96688bf0d4d536adb8eb4ed9bf0002/packages/eslint-config-airbnb/rules/react.js#L47-L57),
+ *
+ *  'react/sort-comp': [2, {
+ *    'order': [
+ *      'lifecycle',
+ *      '/^on.+$/',
+ *      '/^(get|set)(?!(InitialState$|DefaultProps$|ChildContext$)).+$/',
+ *      'everything-else',
+ *      '/^render.+$/',
+ *      'render'
+ *    ]
+ *  }],
+ *
+ * NOTE: only works on React.createClass() syntax, not ES6 class.
+ */
+module.exports = function(fileInfo, api, options) {
+  const j = api.jscodeshift;
+
+  const ReactUtils = require('./utils/ReactUtils')(j);
+
+  const printOptions =
+    options.printOptions || {quote: 'single', trailingComma: true};
+
+  const root = j(fileInfo.source);
+
+  const propertyComparator = (a, b) => {
+    const nameA = a.key.name;
+    const nameB = b.key.name;
+
+    const indexA = getRefPropIndexes(nameA);
+    const indexB = getRefPropIndexes(nameB);
+
+    if (indexA === indexB) {
+      // compare lexically
+      return +(nameA > nameB) || +(nameA === nameB) - 1;
+    } else {
+      // compare by index
+      return indexA - indexB;
+    }
+  };
+
+  const sortComponentProperties = classPath => {
+    const spec = ReactUtils.getReactCreateClassSpec(classPath);
+
+    if (spec) {
+      const orderedProperties = spec.properties.sort(propertyComparator);
```

Comment:
Apologies, it didn't appear to work when I tried it. I see now that it works, and is better because it doesn't make additional changes. On the plus side, while debugging that I caught a different bug. :) (Also fixed)